pycodestyle.py

```diff
@@ -270,7 +271,20 @@ def blank_lines(logical_line, blank_lines, indent_level, line_number,
         if indent_level:
             if not (blank_before or previous_indent_level < indent_level or
                     DOCSTRING_REGEX.match(previous_logical)):
-                yield 0, "E301 expected 1 blank line, found 0"
+                ancestor_level = indent_level
+                nested = False
+                # Search backwards for a def ancestor or tree root (top level).
+                for line in lines[line_number - 2::-1]:
+                    if line.strip() and expand_indent(line) < ancestor_level:
```

Comment:
I decided to do it that way because it keeps the code shorter and the second evaluation it's not very frequent.